### ISSUE TITLE
Renamed folders disappear and folder contents are moved up a directory.

### ISSUE THREAD

**Issue by skunk-ink** [SiaFoundation/renterd#1599]
### Current Behavior

As the title says, when trying to rename a folder in the UI, the folder will disappear, and all the contents of the folder are moved up into the folder's parent. I am unable to grab a screenshot at the moment, so for examples sake:

If I have the directory tree of:
```
TV/
  |- Show 1/
    |- Season 1/
      |- Episode 1.mp4
      |- Episode 2.mp4
      |- Episode 3.mp4
    |-Season 2/
      |- Episode 1.mp4
      |- Episode 2.mp4
      |- Episode 3.mp4
```

And I try to rename the folder `Season 1` to `Season 01`, I end up with a directory tree as follows:
```
TV/
  |- Show 1/
    |- Episode 1.mp4
    |- Episode 2.mp4
    |- Episode 3.mp4
    |-Season 2/
      |- Episode 1.mp4
      |- Episode 2.mp4
      |- Episode 3.mp4
```

My `Season 1` folder no longer appears, and all the episodes it contains have been moved into the TV show's root folder.

### Expected Behavior

`Season 1` should be renamed to `Season 01` and all the episodes it contains should stay within the folder.

### Steps to Reproduce

Rename a folder

### Version

v1.0.8

### What operating system did the problem occur on (e.g. Ubuntu 22.04, macOS 12.0, Windows 11)?

Ubuntu 24.04 Server LTS

### Autopilot Config

{         "contracts": {                 "set": "autopilot",                 "amount": 50,                 "allowance": "7560000000000000000000000000",                 "period": 6048,                 "renewWindow": 2016,                 "download": 1400000000000,                 "upload": 1400000000000,                 "storage": 3000000000000,                 "prune": true         },         "hosts": {                 "allowRedundantIPs": false,                 "maxDowntimeHours": 336,                 "minProtocolVersion": "1.6.0",                 "minRecentScanFailures": 10,                 "scoreOverrides": null         } }

### Bus Config

Contract Set: "default": "autopilot"  Gouging: "hostBlockHeightLeeway": 12, "maxContractPrice": "1000000000000000000000000", "maxDownloadPrice": "1500000000000000000000000000", "maxRPCPrice": "1000000000000000000000", "maxStoragePrice": "92592592593", "maxUploadPrice": "100000000000000000000000000", "migrationSurchargeMultiplier": 10, "minAccountExpiry": 86400000000000, "minMaxEphemeralAccountBalance": "1000000000000000000000000", "minPriceTableValidity": 300000000000  Redundancy: "minShards": 10, "totalShards": 30  Upload Packing: "enabled": true, "slabBufferMaxSizeSoft": 4294967296

### Contract Set Contracts

50

### Anything else?

While filling this out, I noticed my renterd is running at 1.0.8. This is a brand-new setup using our latest docker image. Has the docker image not been updated recently?

**Comment by ChrisSchinnerl** [comment]
@skunk-ink `latest` is only updated upon stable releases. That's `v1.0.8` because `v1.1.0` is still in beta. 

I feel like this issue might have been fixed since then but we will double check

**Comment by skunk-ink** [comment]
Ahh, ok, that makes sense. I didn't realize we skipped a 1.0.9 release.

**Comment by alexfreska** [comment]
@ChrisSchinnerl @peterjan I was able to recreate this in a test, but from what I can tell I don't see an issue in the UI code. Simple example below - maybe the directory empty stub files confuse something?

Starting with following files:
```
/a/
/a/sample.txt
```
Calling with the rename endpoint with:
```
{
  "bucket": "files-test",
  "to": "/b/",
  "from": "/a/",
  "mode": "multi",
  "force": false
}
Results in:
```
/b/
/sample.txt
```

**Comment by alexfreska** [comment]
Here is a video example. I noticed something else strange: deleting the renamed directory still deletes the file that was originally contained.

https://github.com/user-attachments/assets/55b27d08-93c7-4b33-9b31-86c653c05d07

**Comment by rrauch** [comment]
I've noticed this too when upgrading from `1.0.7` (I believe) to `1.0.8`. 

As a workaround I had to switch from the `GET /bus/objects/*key` endpoint to the `POST /bus/objects/list` endpoint and then manually filter out all objects that aren't immediate children of the directory in question.

When trying to find the root cause of this, I noticed a lot of refactoring going on in that area, such as
- https://github.com/SiaFoundation/renterd/pull/1236
- https://github.com/SiaFoundation/renterd/pull/1242

My guess is this issue probably originates from there somewhere.
